base/precompilation.jl

```diff
@@ -489,44 +504,46 @@ function precompilepkgs(pkgs::Vector{String}=String[]; internal_call::Bool=false
     end
     @debug "precompile: circular dep check done"  _group=:precompile
 
-    # if a list of packages is given, restrict to dependencies of given packages
-    if !isempty(pkgs)
-         function collect_all_deps(depsmap, dep, alldeps=Set{Base.PkgId}())
-            for _dep in depsmap[dep]
-                if !(_dep in alldeps)
-                    push!(alldeps, _dep)
-                    collect_all_deps(depsmap, _dep, alldeps)
-                end
-            end
-            return alldeps
-        end
-        keep = Set{Base.PkgId}()
-        for dep in depsmap
-            dep_pkgid = first(dep)
-            if dep_pkgid.name in pkgs
-                push!(keep, dep_pkgid)
-                collect_all_deps(depsmap, dep_pkgid, keep)
+    if isempty(pkgs)
+        pkgs = [pkg.name for pkg in direct_deps]
+        target = "all packages"
+    else
+        target = "project..."
+    end
```

Comment:
Why would the former be "all packages" and the latter be "project..." i.e. all packages?

Reply:
No idea why I changed this. Makes no sense.